components/search_results_item/index.js

```diff
@@ -30,28 +30,33 @@ function mapStateToProps() {
     const getCommentCountForPost = makeGetCommentCountForPost();
 
     return (state, ownProps) => {
-        const config = getConfig(state);
-        const preferences = getMyPreferences(state);
-        const enablePostUsernameOverride = config.EnablePostUsernameOverride === 'true';
         const {post} = ownProps;
-        const user = getUser(state, post.user_id);
-        const channel = getChannel(state, post.channel_id) || {delete_at: 0};
-        const directTeammate = getDirectTeammate(state, channel.id);
+        if (post) {
```

Comment:
Having wildly divergent props based on the use case smells wrong here -- can we break up search results item into two connected components, and render them from a parent that does the right thing instead?

Reply:
Done